### PR TITLE
TestData: Support for csv files & csv content

### PR DESCRIPTION
**Pull request by torkelo**
Goal
Better built-in support for csv test data for visualization test dashboards and demos. 

* Adds CSV File scenario that reads csv files stored in public/testdata 
* Adds CSV Content scenario that reads csv content from query model 
* Removes the categorical test data scenario 
* Updates Bar chart value font-size test dashboard to use new test data scenario 

A lot of overlap between this and the csv input data source. Long term I would like this to be included in the default built-in " -- Grafana --" data source, or be moved to a backend for the input data source. The killer feature would be if you could upload csv files and store the in db or load files from S3/cloud stored csv fils. But leaving that for the future. 

Todo
* Add some interesting csv data 

Dashboard testing value size:
Screenshot from 2021-05-25 19-09-07

### CODE REVIEW

File: pkg/tsdb/testdatasource/csv_data.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse query json %v", err)
+		}
+
+		csvContent := model.Get("csvContent").MustString()
+		alias := model.Get("alias").MustString(q.RefID)
+
+		frame, err := p.loadCsvContent(strings.NewReader(csvContent), alias)
+
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (torkelo) [reply pkg/tsdb/testdatasource/csv_data.go:33]: yes, agree, it was but a space got in there after refactoring

File: pkg/tsdb/testdatasource/csv_data.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+func parseRawStringValues(values []string) interface{} {
+	numericValues := []float64{}
+
+	for _, value := range values {
+		numericValue, err := strconv.ParseFloat(value, 64)
+		if err == nil {
+			numericValues = append(numericValues, numericValue)
+		} else {
+			fmt.Printf("Error reading %v %v", numericValue, err)
```
[the remark this answers is not in the document]
Comment (torkelo) [reply pkg/tsdb/testdatasource/csv_data.go:142]: no this function does not return error, it tries to parse all values as numbers and if all values could be parsed as numbers returns that array, if any fail it returns the orginal string array 
think this was a just a  troubleshooting error log, forgot to remove it

**Comment by ryantxu** [comment]
@torkelo I updated this PR so that it uses the same value > field parsing as the other CSV flavors.  This will now support:
* null values
* boolean
* float64
* strings

There are two snapshot tests mainly to help explore what the results can look like

**Comment by ryantxu** [comment]
I perhaps got carried away :grimacing: the "manual_entry" support has been broken and unusable for a while.  I do not think fixing the UI is worthwhile and everything can be supported via CSV... so this removes manual_entry and migrates it to csv

image

**Comment by torkelo** [comment]
Nice!

**Pull request merged**